### ISSUE TITLE
Don't work with lcobucci/jwt ^3.4 version

### ISSUE THREAD

**Issue by s-prosvirnin** [patrickbussmann/oauth2-apple#24]
`Attempted to call an undefined method named "create" of class "Lcobucci\JWT\Signer\Ecdsa\Sha256". Did you mean to call "createHash"? in /app/vendor/patrickbussmann/oauth2-apple/src/Provider/Apple.php:241`

**Comment by tofra** [comment]
Method create is only available from 4.x

**Comment by s-prosvirnin** [comment]
> Method `create` is only available from 4.x

Yes, but in composer.json is specified "lcobucci/jwt": "^3.4 || ^4.0"

**Comment by Berdir** [comment]
This had been fixed by https://github.com/patrickbussmann/oauth2-apple/pull/20, but then it was changed again in a follow-up commit and is now again different and broken again.

lcobucci/jwt 4 requires PHP 7.4, so that's not always feasible to install.

**Comment by patrickbussmann** [comment]
But thats why we're using this.

    "lcobucci/jwt": "^3.4 || ^4.0"

And the unit tests are running fine via Travis.
https://travis-ci.com/github/patrickbussmann/oauth2-apple

So PHP 5.6 until 8.0 is working fine.

Can you create an example for my which doesnt work?
Because then I could create a test case for it. 😉

**Comment by Berdir** [comment]
I'm not sure how it works, but Lcobucci\JWT\Signer\Ecdsa\Sha256::create() does not exist on 3.4? 

Running the tests locally and checking the coverage report, getConfiguration() is not being tested. Are you mocking that in your test?

**Comment by WouterWidgets** [comment]
I'm getting a slightly different error on `PHP 7.1.33`, sing `patrickbussmann/oauth2-apple:0.2.4`:
```
Call to undefined method Lcobucci\JWT\Signer\Ecdsa\Sha256::create()
```

**Comment by tjveldhuizen** [comment]
That's the same error, in different wording, I think. While stating this package is compatible with lcobucci/jwt ^3.4 in it's package.json, in the code it is not. I've suggested a fix and improved the tests in PR #25

**Comment by wgcvc** [comment]
Hey thanks @tjveldhuizen for the PR
Is there any help needed to get that PR merged/approved?

We got the same problem (just different wording possible due to PHP 5.6 being used):
`PHP Fatal error: Call to undefined method Lcobucci\JWT\Signer\Ecdsa\Sha256::create() in composer/vendor/patrickbussmann/oauth2-apple/src/Provider/Apple.php on line 241`

This started to happen when the package was upgraded from 0.2.2 to 0.2.4 on our side.
